src/Corax/IndexWriter.cs

```diff
@@ -372,6 +371,30 @@ void ExactInsert(ReadOnlySpan<byte> value)
 
                 if (binding.HasSuggestions)
                     AddSuggestions(binding, slice);
+
+                scope?.Dispose();
+            }
+        }
+
+        private static unsafe ByteStringContext<ByteStringMemoryCache>.InternalScope CreateNormalizedTerm(ByteStringContext context, ReadOnlySpan<byte> value, out Slice slice)
+        {
+            ulong hash = 0;
+            int length = value.Length;
+            if (length > Constants.Terms.MaxLength)
+            {
+                int hashStartingPoint = Constants.Terms.MaxLength - 2 * sizeof(ulong);
+                hash = Hashing.XXHash64.Calculate(value.Slice(hashStartingPoint));
```

Comment:
Should we use `xxhash` here? I would feel better if we'll use cyrpto hash here

Reply:
There is no rational reason to pay for the computational cost of a crypto hash in here. The only thing a crypto hash buys you is that it is much harder to generate collisions using reverse engineering of the algorithm. `XXHash` could be subjected to cryptoanalysis tools and reversed and that is why it is not a suitable crypto hash.